assignment6/src/ParkingTest.java

```diff
@@ -0,0 +1,59 @@
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+
+public class ParkingTest {
+    private Parking parking = new Parking();
+
+    @Test
+    public void testNoElements() {
+        List<Movement> movements = parking.RearrangingCars(new int[]{0}, new int[]{0});
+        assertEquals(0, movements.size());
+    }
+
+    @Test
+    public void testOneElementSamePositions() {
+        List<Movement> movements = parking.RearrangingCars(new int[]{0, 1}, new int[]{0, 1});
+        assertEquals(0, movements.size());
+    }
+
+    @Test
+    public void testOneElementDifferentPositions() {
+        List<Movement> movements = parking.RearrangingCars(new int[]{1, 0}, new int[]{0, 1});
+        checkEntries(new int[][]{{0, 1}}, movements);
+    }
+
+    @Test
+    public void testTwoElementsWithTwoMovements() {
+        List<Movement> movements = parking.RearrangingCars(new int[]{2, 1, 0}, new int[]{0, 2, 1});
+        checkEntries(new int[][]{{1, 2}, {0, 1}}, movements);
+    }
+
+    @Test
+    public void testMoreElementsWithoutSamePositionOfZero() {
+        List<Movement> movements = parking.RearrangingCars(new int[]{1, 2, 0, 3}, new int[]{3, 1, 2, 0});
+        checkEntries(new int[][]{{1, 2}, {0, 1}, {3, 0}}, movements);
+    }
+
+    @Test
+    public void testMoreElementsWithSamePositionOfZero() {
+        List<Movement> movements = parking.RearrangingCars(new int[]{1, 2, 0, 3}, new int[]{3, 1, 0, 2});
+        checkEntries(new int[][]{{0, 2}, {3, 0}, {1, 3}, {2, 1}}, movements);
+    }
+
+    private void checkEntries(int[][] realMovements, List<Movement> expectedMovements) {
+        int size = expectedMovements.size();
+        assertThat(size, is(realMovements.length));
+        for (int i = 0; i < size; i++) {
+            Entry<Integer, Integer> entry = expectedMovements.get(i).getValues();
+            assertThat(new int[]{entry.getKey(), entry.getValue()}, is(realMovements[i]));
+        }
+    }
```

Comment:
A better way to check is to run the moves provided on the input provided and check that it results in the expected target array.

Checking that the moves are precisely the moves you expect is brittle - there are multiple sets of moves that produce the correct result, and if someone else tries to run their algorithm against your tests, the algorithm may be fine and the tests fail. This reduces maintainability - if you tweak the algorithm, the tests may break even if there's no real problem. Try to avoid brittle tests.